server/protocols/webadmin/webadmin-data/src/test/java/org/apache/james/webadmin/routes/DomainMappingsRoutesTest.java

```diff
@@ -0,0 +1,387 @@
+            String domain = "from.com";
+
+            when()
+                .get(domain)
+            .then()
+                .contentType(ContentType.JSON)
+                .statusCode(HttpStatus.NOT_FOUND_404)
+                .body("type", is(ErrorResponder.ErrorType.NOT_FOUND.getType()))
+                .body("statusCode", is(HttpStatus.NOT_FOUND_404))
+                .body("message", is("Cannot find mappings for " + domain));
+        }
+
+        @Test
+        void getSpecificDomainMappingShouldRespondWithNotFoundWhenHasEmptyAliases() throws RecipientRewriteTableException {
+            String domain = "from.com";
+
+            Map<MappingSource, Mappings> allMappings = ImmutableMap.of(
+                    MappingSource.fromDomain(Domain.of(domain)),
+                    MappingsImpl.empty()
+            );
+
+            when(recipientRewriteTable.getAllMappings()).thenReturn(allMappings);
+
+            when()
+                .get(domain)
+            .then()
+                .contentType(ContentType.JSON)
+                .statusCode(HttpStatus.NOT_FOUND_404)
+                .body("type", is(ErrorResponder.ErrorType.NOT_FOUND.getType()))
+                .body("statusCode", is(HttpStatus.NOT_FOUND_404))
+                .body("message", is("Cannot find mappings for " + domain));
+        }
+
+        @Test
+        void getSpecificDomainMappingNotDomainsMappingsShouldBeFilteredOut() throws RecipientRewriteTableException {
+            String domain = "from.com";
+            String aliasDomain = "to.com";
+            final MappingSource mappingSource = MappingSource.fromDomain(Domain.of(domain));
+
+            recipientRewriteTable.addRegexMapping(mappingSource, "(.*)@localhost");
+            recipientRewriteTable.addGroupMapping(mappingSource, "user@domain.com");
+            recipientRewriteTable.addForwardMapping(mappingSource, "user@domain.com");
+            recipientRewriteTable.addErrorMapping(mappingSource, "disabled");
+            recipientRewriteTable.addAliasDomainMapping(mappingSource, Domain.of(aliasDomain));
+
+            List<String> body =
+                    when()
+                        .get(domain)
+                    .then()
+                        .contentType(ContentType.JSON)
+                        .statusCode(HttpStatus.OK_200)
+                    .extract()
+                        .jsonPath()
+                        .getList(".");
+
+            assertThat(body).containsOnly(aliasDomain);
+        }
+
+        @Test
+        void getSpecificDomainMappingShouldResponseWithOK() throws RecipientRewriteTableException {
+            String domain = "abc.com";
+            String aliasDomain = "a.com";
+            MappingSource mapping = MappingSource.fromDomain(Domain.of(domain));
+            Map<MappingSource, Mappings> mappings = ImmutableMap.of(mapping, MappingsImpl.fromRawString("domain:" + aliasDomain));
```

Comment:
Can't you use MappingsImpl.fromMappings(Stream.of(Mapping.domain(aliasDomain))) ?

We might want to add in MappingsImpl a convenience `MappingsImpl fromMappings(Mapping... mappings) ` if you find it easier...